app/controllers/sys_controller.rb

```diff
@@ -99,14 +101,37 @@ def check_enabled
   private
 
   def require_basic_auth
+    @authenticated_user =
+    if Setting.repository_authentication_bypass?
+      find_user_from_basic_auth
+    else
+      # Authenticate the user internally
+      perform_basic_auth
+    end
+
+    render_authorization_required unless @authenticated_user.present?
```

Comment:
Oh this is why I hate certain shortcuts in ruby syntax xD

I simply did not see line 104. Thus I assumed lines 105-110 would be a simple if-else-statement that assigns nothing to anything.

But I also have no taste when it comes to ruby, because I like languages that contain braces `{ }` ;-)

I'd like to duplicate the assignment here, but lets ask people in whom I trust when it comes to code style:

@Azure7111 @machisuji I summon you to tell me what you think about this assignment: ![summoning](http://upload.wikimedia.org/wikipedia/en/b/bd/ThePlaneteers.JPG)